projects/packages/my-jetpack/_inc/state/actions.js

```diff
@@ -109,6 +111,7 @@ function requestProductStatus( productId, data, { select, dispatch } ) {
 			.then( freshProduct => {
 				dispatch( setIsFetchingProduct( productId, false ) );
 				dispatch( setProduct( freshProduct ) );
+				globalDispatch( CONNECTION_STORE_ID ).refreshConnectedPlugins();
```

Comment:
I think we can access to other stores by taking advantage of [registry](https://developer.wordpress.org/block-editor/how-to-guides/thunks/#registry) feature, supported by the API and avoiding to call the dispatch function.

Reply:
https://github.com/Automattic/jetpack/pull/23358/commits/72e171307152633ba787ebbe03457faab0b6222c